internal/controller/nnf_node_block_storage_controller.go

```diff
@@ -688,16 +996,71 @@ func (r *NnfNodeBlockStorageReconciler) NnfEcEventEnqueueHandler(ctx context.Con
 	return requests
 }
 
+// checkComputeFenced checks if a fence response file exists for the given compute node,
+// indicating it has been fenced and should not have storage groups restored
+func (r *NnfNodeBlockStorageReconciler) checkComputeFenced(computeName string, log logr.Logger) (bool, error) {
+	// Check if fence response directory exists
+	if _, err := os.Stat(fence.ResponseDir); os.IsNotExist(err) {
+		return false, nil
+	}
+
+	// Read all files in the response directory
+	entries, err := os.ReadDir(fence.ResponseDir)
+	if err != nil {
+		return false, err
+	}
+
+	// Check each response file to see if it's for this compute node
+	for _, entry := range entries {
+		if entry.IsDir() || filepath.Ext(entry.Name()) != ".json" {
+			continue
+		}
+
+		responseFile := filepath.Join(fence.ResponseDir, entry.Name())
+		data, err := os.ReadFile(responseFile)
+		if err != nil {
+			log.Error(err, "Failed to read fence response file", "file", responseFile)
+			continue
+		}
+
+		var response struct {
+			TargetNode string `json:"target_node"`
+			Success    bool   `json:"success"`
+		}
+
+		if err := json.Unmarshal(data, &response); err != nil {
+			log.Error(err, "Failed to parse fence response file", "file", responseFile)
+			continue
+		}
+
+		// If this response is for our compute node and the fence was successful, it's fenced
+		if response.TargetNode == computeName && response.Success {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *NnfNodeBlockStorageReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err := mgr.Add(r); err != nil {
 		return err
 	}
 
 	// nnf-ec is not thread safe, so we are limited to a single reconcile thread.
-	return ctrl.NewControllerManagedBy(mgr).
+	builder := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
 		For(&nnfv1alpha9.NnfNodeBlockStorage{}).
-		WatchesRawSource(&source.Channel{Source: r.Events}, handler.EnqueueRequestsFromMapFunc(r.NnfEcEventEnqueueHandler)).
-		Complete(r)
+		WatchesRawSource(&source.Channel{Source: r.Events}, handler.EnqueueRequestsFromMapFunc(r.NnfEcEventEnqueueHandler))
+
+	// Watch fence events from NnfNode (only if channel was provided)
+	if r.FenceEvents != nil {
+		r.Log.Info("Watching FenceEvents channel")
+		builder = builder.WatchesRawSource(&source.Channel{Source: r.FenceEvents}, &handler.EnqueueRequestForObject{})
```

Comment:
Does `handler.EnqueueRequestForObject{}` here generate a reconcile for each NnfNodeBlockStorage that exists?